source/common/access_log/access_log_formatter.cc

```diff
@@ -150,23 +161,13 @@ RequestInfoFormatter::RequestInfoFormatter(const std::string& field_name) {
     };
   } else if (field_name == "REQUEST_DURATION") {
     field_extractor_ = [](const RequestInfo::RequestInfo& request_info) {
-      Optional<std::chrono::microseconds> duration = request_info.requestReceivedDuration();
-      if (duration.valid()) {
-        return std::to_string(
-            std::chrono::duration_cast<std::chrono::milliseconds>(duration.value()).count());
-      } else {
-        return UnspecifiedValueString;
-      }
+      return AccessLogFormatUtils::durationToString(request_info,
+                                                    request_info.lastDownstreamRxByteReceived());
     };
   } else if (field_name == "RESPONSE_DURATION") {
     field_extractor_ = [](const RequestInfo::RequestInfo& request_info) {
-      Optional<std::chrono::microseconds> duration = request_info.responseReceivedDuration();
-      if (duration.valid()) {
-        return std::to_string(
-            std::chrono::duration_cast<std::chrono::milliseconds>(duration.value()).count());
-      } else {
-        return UnspecifiedValueString;
-      }
+      return AccessLogFormatUtils::durationToString(request_info,
+                                                    request_info.firstUpstreamRxByteReceived());
```

Comment:
For RESPONSE_DURATION, shouldn't this be looking at when the last Upstream byte was received?

Reply:
RESPONSE_DURATION was measured off of the old `responseReceived` value, which was captured in [`onUpstreamHeaders`](https://github.com/envoyproxy/envoy/pull/2660/files#diff-a00f27eaffcd28ed6b74102912eb1d65L629), so it measured TTFB, not the complete response. Unless I'm misinterpreting this...